### PR TITLE
Adds dart:convert code.

### PR DESCRIPTION
**Pull request by redbrogdon**
Adds the tab pages for `dart:convert`-based JSON deserialization to the example.

### CODE REVIEW

File: lib/dart_convert/converted_complex_object.dart [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      aListOfStrings: json['aListOfStrings'] != null ?
+          List<String>.from(json['aListOfStrings']) : null,
+      aListOfInts: json['aListOfInts'] != null ?
+          List<int>.from(json['aListOfInts']) : null,
+      aListOfDoubles: json['aListOfDoubles'] != null ?
+          List<double>.from(json['aListOfDoubles']) : null,
+      aListOfObjects: objects,
+    );
+  }
+}
```
Comment (kevmoo) [inline lib/dart_convert/converted_complex_object.dart]: run dartfmt – missing new line

File: lib/dart_convert/converted_complex_object.dart [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  final double aDouble;
+  final ConvertedSimpleObject anObject;
+  final List<String> aListOfStrings;
+  final List<int> aListOfInts;
+  final List<double> aListOfDoubles;
+  final List<ConvertedSimpleObject> aListOfObjects;
+
+  factory ConvertedComplexObject.fromJson(Map<String, dynamic> json) {
+    if (json == null)
+      return null;
```
Comment (kevmoo) [inline lib/dart_convert/converted_complex_object.dart]: dartfmt would fix this, too. We don't like hanging implicit blocks – either one line or add `{}`

File: lib/utils.dart [hunk 3]
```diff
@@ -0,0 +1,28 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+String prettyPrintList(Iterable iter) {
+  if (iter == null) return 'NULL';
+
+  if (iter.length == 0) return '[]';
```
Comment (kevmoo) [inline lib/utils.dart]: Not needed. Your code below will handle the empty case fine.

Also, prefer `isEmpty`

**Review by kevmoo — APPROVED** [review]
Some nits.

I'd recommed adding analysis options, too..

File: lib/dart_convert/converted_complex_object.dart [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      anObject: json['anObject'] != null
+          ? ConvertedSimpleObject.fromJson(json['anObject'])
+          : null,
+      aListOfStrings: json['aListOfStrings'] != null ?
+          List<String>.from(json['aListOfStrings']) : null,
+      aListOfInts: json['aListOfInts'] != null ?
+          List<int>.from(json['aListOfInts']) : null,
+      aListOfDoubles: json['aListOfDoubles'] != null ?
+          List<double>.from(json['aListOfDoubles']) : null,
+      aListOfObjects: objects,
```
Comment (kevmoo) [inline lib/dart_convert/converted_complex_object.dart]: it's a bit weird that you choose to not inline the `ConvertedSimpleObject` here, too...

File: lib/dart_convert/converted_complex_object.dart (hunk 1 above)
Comment (redbrogdon) [reply lib/dart_convert/converted_complex_object.dart]: Nice catch, thanks. Fixed.

File: lib/dart_convert/converted_complex_object.dart (hunk 2 above)
Comment (redbrogdon) [reply lib/dart_convert/converted_complex_object.dart]: Fixed.

File: lib/utils.dart (hunk 3 above)
Comment (redbrogdon) [reply lib/utils.dart]: TIL about isEmpty. Fixed.

File: lib/dart_convert/converted_complex_object.dart (hunk 4 above)
Comment (redbrogdon) [reply lib/dart_convert/converted_complex_object.dart]: I think that bit may have been more complicated in an earlier revision, so I separated it out.

Changed it to inline now, though.

**Pull request merged**